proposals/default-interface-methods.md

```diff
@@ -1,43 +1,227 @@
+
+> ***Open Issue:*** Should operator declarations be permitted in an interface? Probably not conversion operators, but what about others?
+
+> ***Open Issue:*** Should `new` be permitted on interface member declarations that hide members from base interfaces?
+
+> ***Open Issue:*** Should `const` declarations be permitted in an interface?
+
+
+
+
+### Overrides in interfaces
+
+Override declarations (i.e. those containing the `override` modifier) allow the programmer to provide a most specific implementation of a virtual member in an interface where the compiler or runtime would not otherwise find one. It also allows turning an abstract member from a super-interface into a default member in a derived interface. An override declaration is permitted to *explicitly* override a particular base interface method by qualifying the declaration with the interface name (no access modifier is permitted in this case).
+
+
+``` c#
+interface IA
+{
+    void M() { WriteLine("IA.M"); }
+}
+interface IB : IA
+{
+    override void IA.M() { WriteLine("IB.M"); } // explicitly named
+}
+interface IC : IA
+{
+    override void M() { WriteLine("IC.M"); } // implicitly named
+}
+```
+
+If the interface is not named in the override declaration, then all matching methods (from direct or indirect base interfaces) are overridden. There must be at least one such method or the override declaration is an error.
+
+> ***Open issue***: should that "direct and indirect" be "direct" here?
+
+Overrides in interfaces are useful to provide a more specific (e.g. more efficient) implementation of a base interface's method. For example, a new `First()` method on `IEnumerable` may have a much more efficient implementation on the interface `IList`.
+
+A method declared in an interface is never treated as an `override` of another method unless it contains he `override` modifier. This is necessary for compatibility.
+
+``` c#
+interface IA
+{
+    void M();
+}
+interface IB : IA
+{
+    void M(); // not related to 'IA.M'; not an override
+}
+```
+
+Override declarations in interfaces may not be declared `sealed`.
+
+
+### Reabstraction
+
+A virtual (concrete) method declared in an interface may be overridden to be abstract in a derived interface
+
+``` c#
+interface IA
+{
+    void M() { WriteLine("IA.M"); }
+}
+interface IB : IA
+{
+    override abstract void M();
+}
+class C : IB { } // error: class 'C' does not implement 'IA.M'.
+```
+
+The `abstract` modifier is not required in the declaration of `IB.M` (that is the default in interfaces), but it is probably good practice to be explicit in an override declaration.
+
+This is useful in derived interfaces where the default implementation of a method is inappropriate and a more appropriate implementation should be provided by implementing classes.
+
+> ***Open Issue:*** Should reabstraction be permitted?
```

Comment:
We discussed this and decided "NO", however I've reopened this as the decision contradicts the goals of the feature, as described in https://github.com/dotnet/csharplang/issues/406